packaging/setup/plugins/ovirt-engine-setup/ovirt-engine/network/ovirtproviderovn.py

```diff
@@ -1088,13 +1097,18 @@ def _misc_configure_ovn_timeout(self):
             oenginecons.Stages.CA_AVAILABLE,
             oenginecons.Stages.OVN_SERVICES_RESTART,
         ),
-        condition=lambda self:
-            self._enabled
     )
     def _misc_configure_provider(self):
-        self._generate_client_secret()
-        self._configure_ovirt_provider_ovn()
-        self._upate_external_providers_keystore()
+        if self._enabled:
+            self._generate_client_secret()
+            self._configure_ovirt_provider_ovn()
+            self._upate_external_providers_keystore()
+        elif self._provider_installed:
+            keycloak_enabled = self.environment.get(
+                oengcommcons.KeycloakEnv.KEYCLOAK_ENABLED
+            )
+            if keycloak_enabled:
```

Comment:
Same

Reply:
In this case, when subsequent engine-setup is executed on environment with already enabled keycloak, the ovirt provider ovn config `/etc/ovirt-provider-ovn/conf.d/10-setup-ovirt-provider-ovn.conf` will be simply rewritten. This is not the prettiest approach, but it is harmless and I think I saw similar cases elsewhere in setup code.  No additional credentials related questions are supposed to be asked.

Reply:
You are right that there are many other such cases, as I noted in the dwh patch. It will not even be rewritten, because filetransaction won't rewrite it if it already has the content we want. And the secret key is saved in postinstall, and AFAICT with this the content indeed will be the same. Didn't test.

However, if you do accept my suggestion to add a var CONFIGURED, perhaps better use it here too. Or not, if you think it's better to overwrite if needed. In the original place we added CONFIGURED, we actually eventually had to partially ignore it, because we wanted to make further changes to the ssl conf file (and later also had to revert that, due to wanting to rely on crypto-policies, but that's a different story...).